agent/pipeline_parser.go

```diff
@@ -57,26 +57,51 @@ func (p PipelineParser) Parse() (pipeline interface{}, err error) {
 	return interpolated, nil
 }
 
-func (p PipelineParser) interpolateEnvBlock(envMap map[string]interface{}) error {
-	// do a first pass without interpolation
-	for k, v := range envMap {
-		switch tv := v.(type) {
-		case string, int, bool:
-			p.Env.Set(k, fmt.Sprintf("%v", tv))
+func (p PipelineParser) interpolateTopLevelEnvBlock(envMap map[string]interface{}) error {
+	graph := toposort.NewGraph(10)
```

Comment:
10?

Reply:
Yeah, it's a initial size for the graph. Probably merits a comment.